packages/fetchai/connections/gym/tests/test_gym.py

```diff
@@ -0,0 +1,279 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#   Copyright 2018-2021 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the tests of the gym connection module."""
+import asyncio
+import logging
+import os
+from typing import cast
+from unittest.mock import MagicMock, patch
+
+import gym
+import pytest
+
+from aea.common import Address
+from aea.configurations.base import ConnectionConfig
+from aea.identity.base import Identity
+from aea.mail.base import Envelope, Message
+from aea.protocols.dialogue.base import Dialogue as BaseDialogue
+
+from packages.fetchai.connections.gym.connection import GymConnection
+from packages.fetchai.protocols.gym.dialogues import GymDialogue
+from packages.fetchai.protocols.gym.dialogues import GymDialogues as BaseGymDialogues
+from packages.fetchai.protocols.gym.message import GymMessage
+
+from tests.conftest import ROOT_DIR, UNKNOWN_PROTOCOL_PUBLIC_ID
```

Comment:
So this would have to change right. Because now this is ill-defined.

Reply:
correct